src/core/include/openvino/op/pad.hpp

```diff
@@ -47,39 +48,52 @@ class OPENVINO_API Pad : public Op {
     /// \param pad_mode The padding mode: CONSTANT, EDGE, REFLECT or SYMMETRIC.
     Pad(const Output<Node>& arg, const Output<Node>& pads_begin, const Output<Node>& pads_end, PadMode pad_mode);
 
-    /// \brief Constructs a generic padding operation.
+    std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override;
+    bool has_evaluate() const override;
```

Comment:
Can be `has_evaluate` and `evaluate` part of base class?
It looks like they are same and could be shared as they call common protected function.

Reply:
This is a special case because there is no difference in the evaluate implementation between Pad-1 and Pad-12, 
but in general evaluate is related to the specific operation not with the Base class.
Overall `evaluate` can be a part of the PadBase class, and it was in the initial commits, 
but has been moved after discussion with @tomdol.